packages/shared/src/hooks/useTutorial.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useMemo } from 'react';
+import { useQuery, useQueryClient } from 'react-query';
+import usePersistentContext from './usePersistentContext';
+
+export enum TutorialKey {
+  SEEN_NEW_SQUAD_TOOLTIP_KEY = 'seenNewSquadTooltip',
+}
+
+export type UseTutorialProps = {
+  key: TutorialKey;
+  defaultState?: boolean;
+};
+
+export type UseTutorial = {
+  isActive: boolean;
+  activate: () => void;
+  isCompleted: boolean;
+  complete: (active?: boolean) => Promise<void>;
+  reset: () => Promise<void>;
+  isFetched: boolean;
+};
+
+export const useTutorial = ({
+  key,
+  defaultState = false,
+}: UseTutorialProps): UseTutorial => {
+  const client = useQueryClient();
+  const [isCompleted, setCompleted, isFetched] = usePersistentContext(
+    `tutorial-${key}`,
+    false,
+  );
+
+  const queryKey = ['tutorial', 'query', key];
```

Comment:
Why do we mix two keys though?
Can't we leverage the persistent context to keep track of the `isActive` and `setActive` as well?

Reply:
Hmm good question can you raise it in the `#feat-squads-closedbetav3` channel for Sab?

Reply:
Going without persisting active state for now, [thread here](https://dailydotdev.slack.com/archives/C04UED2504R/p1679490131404769)

Reply:
Ok, but would it not be simpler from a engineering side of view?
Now you have 2 queries basically doing the exact same thing already.

But maybe I'm missing something here.
Happy to chat about it.

Reply:
I mean can't we simply have default = false, and if it's true it means it's active?
Then turn it to false once it's completed again?

Not sure we need the extra state with react query if that makes sense.

Reply:
I did think about it, how would you differentiate between activated and completed then?

Let's take a look at carousel example. So, we need to add animation around post field when we are on one of the carousel slides. If we go with two states we have (I am gonna show the state in each step as `{ isActive, isCompleted }`:
- users opens the carousel for the first time, state is `{ isActive: false, isCompleted: false }`
- user opens the target carousel slide
  - we activate the slide because tutorial is not yet completed `{ isActive: true, isCompleted: false }` 
- user closes the carousel
  - we finish the tutorial `{ isActive: false, isCompleted: true }`
- user opens the carousel again `{ isActive: false, isCompleted: true }`
- user opens to target carousel slide
  - we know that tutorial has been completed so we do not activate again  `{ isActive: false, isCompleted: true }`
- user closes the carousel
  - we know that tutorial has been completed so we don't need to complete again  `{ isActive: false, isCompleted: true }`
  
If we would to have single state eg. `{ isActive }`
- users opens the carousel for the first time, state is `{ isActive: false }`
- user opens the target carousel slide
  - we activate the slide because tutorial is not yet completed `{ isActive: true }` 
- user closes the carousel
  - we finish the tutorial `{ isActive: false }`
- user opens the carousel again `{ isActive: false }`
- user opens to target carousel slide
  - we just now carousel is not active so we activate it again  `{ isActive: true }`
- user closes the carousel
  -  we complete it again  `{ isActive: false }`

No matter how many times we run it we would always show the tutorial in this case.
  
So basically I think 2 states are needed just because we have an arbitrary completion conditions. If complete conditions was something that we could express in another state eg. we don't show animation anymore if user created at least one post inside squad that would eliminate the need for two states. Does this make sense, if I am missing something I would be glad to simplify this 🙏